domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/casetype/CaseTypeService.java

```diff
@@ -12,6 +12,8 @@
 
     void createAll(JurisdictionEntity jurisdiction, Collection<CaseTypeEntity> caseTypes);
 
+    List<CaseType> findAll();
```

Comment:
Are we retrieving all case types to do an exist? Isn't not performant and has high memory consumption? 
Can't we just ask the db whether a case type with a certain id already exists? (a count query on the db)

Reply:
We need all jurisdictions from the existing case types as we need to make sure we are not importing the same case type with different jurisdiction.

This is on import only so it should not affect performance that much.

Reply:
We are retrieving all most recent versions of case types here.

Reply:
Ok, to issue such count call I would have to make a call for each iteration of case types (see CaseTypeServiceImpl) hence it would be multiple calls (depending on how many case types there is defined in spreadsheet) instead of just one call at the beginning to get all.

Reply:
that is ok, because a count query is very efficient. So even if you endup with 3 count queries one for each imported case type is always way more efficient than retrieving all the latest versions of all the case types

Reply:
As a compromise, if we really want to take a shortcut, I think you can just use the following method we already have in the repository:
```    Optional<CaseTypeEntity> findCurrentVersionForReference(@Param("caseTypeReference") String caseTypeReference);
```
but this is still not as efficient as an exist method which returns a boolean, which is what I'm suggesting. I would avoid even the above method, for performance reasons

Reply:
done